src/stores/HistoryStore.js

```diff
@@ -0,0 +1,239 @@
+import { observable, action, reaction, runInAction } from 'mobx';
+import { filter } from 'lodash';
+import { TransactionStatus, Routes } from 'constants';
+import { transactions } from '../network/graphql/queries';
+
+const EVENT_HISTORY_LIMIT = 5;
+const ACTIVITY_HISTORY_LIMIT = 10;
+const INIT_VALUES = {
+  transactions: [],
+  myTransactions: [],
+  skip: 0, // skip for transaction
+  loaded: false,
+  loadingMore: false,
+  hasMore: true,
+  eventSkip: 0, // skip when query events
+  betSkip: 0,
+  resultSetSkip: 0,
+  withdrawSkip: 0,
+  limit: 0,
+};
+
+export default class {
+  @observable transactions = INIT_VALUES.transactions; // Full txs list fetched by batches
+  @observable myTransactions = INIT_VALUES.myTransactions; // Full txs list fetched by batches
+  @observable loadingMore = INIT_VALUES.loadingMore;
+  @observable loaded = INIT_VALUES.loaded;
+  @observable hasMore = INIT_VALUES.hasMore;
+  @observable myHasMore = INIT_VALUES.hasMore;
+  limit = INIT_VALUES.limit;
+  // for All Transaction in Event page and in Activity history page
+  skip = INIT_VALUES.skip;
+  eventSkip = INIT_VALUES.eventSkip;
+  betSkip = INIT_VALUES.betSkip;
+  resultSetSkip = INIT_VALUES.resultSetSkip;
+  withdrawSkip = INIT_VALUES.withdrawSkip;
+  // for My Transaction in Event page
+  mySkip = INIT_VALUES.skip;
+  myEventSkip = INIT_VALUES.eventSkip;
+  myBetSkip = INIT_VALUES.betSkip;
+  myResultSetSkip = INIT_VALUES.resultSetSkip;
+  myWithdrawSkip = INIT_VALUES.withdrawSkip;
+
+  constructor(app) {
+    this.app = app;
+
+    // Wallet addresses changed
+    reaction(
+      () => this.app.naka.account,
+      () => {
+        if (this.app.ui.location === Routes.ACTIVITY_HISTORY) {
```

Comment:
do you need to react on the Event detail page too?

Reply:
Good catch